src/Components/Endpoints/src/Discovery/PageCollectionBuilder.cs

```diff
@@ -0,0 +1,153 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Represents the list of pages in a <see cref="ComponentApplicationBuilder"/>.
+/// </summary>
+internal class PageCollectionBuilder
```

Comment:
Could you clarify why we need ComponentCollectionBuilder and PageCollectionBuilder as independent implementations, with one of them public and one internal?

Reply:
Sorry, my mistake. I was mixed up between `ComponentBuilder` and `ComponentCollectionBuilder` and `ComponentLibraryBuilder`.